indico/modules/events/api.py

```diff
@@ -255,8 +273,243 @@ def _iterate_objs(objIds):
                 obj = ch.getById(objId, True)
                 if obj is not None:
                     yield obj
+        return self.serialize_events(events)
 
-        return self._process(_iterate_objs(idlist))
+    def serialize_events(self, events):
+        return map(self._build_event_api_data, events)
+
+    def _serialize_date(self, date):
+        if date:
+            return {
+                'date': str(date.date()),
+                'time': str(date.time()),
+                'tz': str(date.tzinfo)
+            }
+
+    def _serialize_persons(self, persons, person_type):
+        return [self._serialize_person(person, person_type) for person in persons]
+
+    def _serialize_person(self, person, person_type):
+        if person:
+            return {
+                '_type': person_type,
+                '_fossil': self.fossils_mapping['person'].get(person_type, None),
+                'fullName': person.full_name,
+                'id': person.id if getattr(person, 'id', None) else person.person_id,
+                'affiliation': person.affiliation,
+                'emailHash': md5(person.email).hexdigest() if person.email else None,
+            }
+
+    def _serialize_convener(self, convener):
+        return {
+            'fax': '',
+            'familyName': convener.last_name,
+            'name': convener.full_name,
+            'firstName': convener.first_name,
+            'phone': convener.phone,
+            'title': convener.title,
+            '_type': 'SlotChair',
+            'email': convener.person.email,
+            'affiliation': convener.affiliation,
+            'address': convener.address,
+            '_fossil': 'conferenceParticipation',
+            'fullName': convener.full_name,
+            'id': convener.person_id
+        }
+
+    def _serialize_session(self, session):
+        return {
+            'folders': build_folders_api_data(session),
+            # TODO: Add when #2272 gets merged
+            # 'startDate': self._serialize_date(session.start_dt) if session.blocks else None,
+            # 'endDate': self._serialize_date(session.end_dt) if session.blocks else None,
+            '_type': 'Session',
+            'sessionConveners': [self._serialize_convener(convener) for convener in session.conveners],
+            'title': session.title,
+            'color': '#{}'.format(session.colors.background),
+            'textColor': '#{}'.format(session.colors.text),
+            'description': session.description,
+            'material': build_material_legacy_api_data(session),
+            'isPoster': session.is_poster,
+            'url': url_for('event.sessionDisplay', confId=session.event_id, sessionId=session.friendly_id,
+                           _external=True),  # TODO: Change to new endpoint once it's there
+            'protectionURL': url_for('sessions.session_protection', session.event_new, session, _external=True),
```

Comment:
not sure if keeping this URL makes any sense; the page is not meant to be loaded outside a dialog. maybe better to remove it?